File: backend/src/venues/dto/update-venue.dto.ts
import {
  IsString,
  IsOptional,
  IsArray,
  IsNumber,
} from 'class-validator';
import { Type } from 'class-transformer';
import { ApiProperty } from '@nestjs/swagger';

export class UpdateVenueDto {
  @ApiProperty({ required: false })
  @IsString()
  @IsOptional()
  name?: string;

  @ApiProperty({ required: false })
  @IsString()
  @IsOptional()
  address?: string;

  @ApiProperty({ required: false })
  @IsString()
  @IsOptional()
  city?: string;

  @ApiProperty({ required: false })
  @IsNumber()
  @IsOptional()
  @Type(() => Number)
  lat?: number;

  @ApiProperty({ required: false })
  @IsNumber()
  @IsOptional()
  @Type(() => Number)
  lng?: number;

  @ApiProperty({ required: false })
  @IsString()
  @IsOptional()
  description?: string;

  @ApiProperty({ required: false })
  @IsArray()
  @IsString({ each: true })
  @IsOptional()
  photos?: string[];
}

File: boxgaming/lib/features/venues/dto/update_venue_dto.dart
class UpdateVenueDto {
  final String? name;
  final String? address;
  final String? city;
  final double? lat;
  final double? lng;
  final String? description;
  final List<String>? photos;

  UpdateVenueDto({
    this.name,
    this.address,
    this.city,
    this.lat,
    this.lng,
    this.description,
    this.photos,
  });

  Map<String, dynamic> toJson() {
    final map = <String, dynamic>{};
    if (name != null) map['name'] = name;
    if (address != null) map['address'] = address;
    if (city != null) map['city'] = city;
    if (lat != null) map['lat'] = lat;
    if (lng != null) map['lng'] = lng;
    if (description != null) map['description'] = description;
    if (photos != null) map['photos'] = photos;
    return map;
  }
}

